examples/1-execute-query-mutation/src/Previewer.re

```diff
@@ -101,117 +101,108 @@ type state = {
-    };
-  },
-  render: self => {
-    <div className=Styles.wrapper>
-      <div className=Styles.side>
-        <section className=Styles.section>
-          <span className=Styles.title> "Query"->str </span>
-          <div className={Styles.colors.query->Styles.code}>
-            {|query dogs {
-  dogs {
-    name
-    breed
-    likes
-  }
-}|}->str
-          </div>
-          <button
-            className={Styles.colors.query->Styles.button}
-            onClick={_event => self.send(ExecuteQuery)}>
-            "Execute Query"->str
-          </button>
-        </section>
-        {switch (String.length(self.state.query)) {
-         | 0 => ReasonReact.null
-         | _ =>
-           <section className=Styles.section>
-             <span className=Styles.title> "Result"->str </span>
-             <div className={Styles.colors.query->Styles.code}>
-               self.state.query->str
-             </div>
-             <button
-               className={Styles.colors.query->Styles.button}
-               onClick={_event => self.send(ClearQuery)}>
-               "Clear Query"->str
-             </button>
-           </section>
-         }}
-      </div>
-      <div className=Styles.side>
-        <section className=Styles.section>
-          <span className=Styles.title> "Mutation"->str </span>
-          <div className={"db4d3f"->Styles.code}>
-            {|mutation likeDog($key: ID!) {
+let initialState = {query: "", mutation: ""};
+
+[@react.component]
+let make = () => {
+  let (state, dispatch) =
+    React.useReducer(
+      (state, action) =>
+        switch (action) {
+        | SetQuery(query) => {...state, query}
+        | ClearQuery => {...state, query: ""}
+        | SetMutation(mutation) => {...state, mutation}
+        | ClearMutation => {...state, mutation: ""}
+        },
+      initialState,
+    );
+
+  let executeQuery = () =>
+    Client.executeQuery(~client, ~query=queryRequest, ())
+    |> Wonka.forEach((. response) =>
+         dispatch(SetQuery(Js.Json.stringifyWithSpace(response##data, 2)))
+       );
+
+  let executeMutation = () =>
+    Client.executeMutation(~client, ~mutation=mutationRequest, ())
+    |> Wonka.forEach((. response) =>
+         dispatch(
+           SetMutation(Js.Json.stringifyWithSpace(response##data, 2)),
+         )
+       );
+  <div className=Styles.wrapper>
+    <div className=Styles.side>
+      <section className=Styles.section>
+        <span className=Styles.title> "Query"->str </span>
+        <div className={Styles.colors.query->Styles.code}>
+          {|query dogs {
+            dogs {
+              name
+              breed
+              likes
+            }
+          }|}
+          ->str
+        </div>
+        <button
+          className={Styles.colors.query->Styles.button}
+          onClick={_event => executeQuery()}>
+          "Execute Query"->str
+        </button>
+      </section>
+      {switch (String.length(state.query)) {
+       | 0 => ReasonReact.null
```

Comment:
While we're here would you mind switching this to `React.null`? Just want to get rid of all `ReasonReact` references while this is fresh ❤️

Reply:
Also changed a few `ReasonReact.string` to `Reason.string`. Unless you didn't want that changed?